mq.go

```diff
@@ -74,8 +82,10 @@ func New(config Config) (MQ, error) {
 		internalErrorChannel: make(chan error),
 		consumers:            newConsumersRegistry(len(config.Consumers)),
 		producers:            newProducersRegistry(len(config.Producers)),
+		state:                new(int32),
 	}
 
+	atomic.StoreInt32(mq.state, ConnectionStateConnecting)
```

Comment:
Connection state management should be encapsulated in the `connect` method.
The default value for state flag should be default value of the respective type.